staging/src/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go

```diff
@@ -231,7 +238,16 @@ func buildImpersonationRequests(headers http.Header) ([]v1.ObjectReference, erro
 		}
 	}
 
-	if (hasGroups || hasUserExtra) && !hasUser {
+	requestedUID := headers.Get(authenticationv1.ImpersonateUIDHeader)
+	hasUID := len(requestedUID) > 0
+	if hasUID {
+		impersonationRequests = append(impersonationRequests, v1.ObjectReference{
```

Comment:
Mentioned earlier, but I think we want this to be `APIVersion: authenticationv1.SchemeGroupVersion.String(),`

@deads2k is this comment from above still valid?  What internal version?

```
// we only parse out a group above, but the parsing will fail if there isn't SOME version
// using the internal version will help us fail if anyone starts using it
```